Sources/StreamChat/StateLayer/DatabaseObserver/StateLayerDatabaseObserver.swift

```diff
@@ -52,14 +57,27 @@ final class StateLayerDatabaseObserver<ResultType: DatabaseObserverType, Item, D
 
 @available(iOS 13.0, *)
 extension StateLayerDatabaseObserver where ResultType == EntityResult {
+    convenience init(
+        databaseContainer: DatabaseContainer,
+        fetchRequest: NSFetchRequest<DTO>,
+        itemCreator: @escaping (DTO) throws -> Item
+    ) {
```

Comment:
Created separate init methods since list vs entity observing uses different arguments. Sorting does not make sense for single entity observing.